src/sec_edgar_bulker/core/utils.py

```
@@ -0,0 +1,168 @@
+                for rotated_file in Path(base_dir).glob(f"{base_name}.*"):
+                    os.chmod(rotated_file, stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IROTH)
+        
+        return logger
+        
+    except Exception as e:
+        # Ensure we have at least console logging in case of errors
+        console_handler = logging.StreamHandler()
+        console_handler.setFormatter(logging.Formatter('%(levelname)s: %(message)s'))
+        logger.addHandler(console_handler)
+        logger.setLevel(logging.ERROR)
+        logger.error(f"Failed to setup logging: {e}")
```

Ruff (0.8.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)